code/controllers/subsystem/vote.dm

```diff
@@ -240,121 +236,103 @@ SUBSYSTEM_DEF(vote)
 		return TRUE
 	return FALSE
 
-
-/datum/controller/subsystem/vote/proc/interface(client/C)
-	if(!C)
-		return
-	var/admin = FALSE
-	if(check_other_rights(C, R_ADMIN, FALSE))
-		admin = TRUE
-	voting |= C
-
-	if(mode)
-		if(question)
-			. += "<h2>Vote: '[question]'</h2>"
-		else
-			. += "<h2>Vote: [capitalize(mode)]</h2>"
-		. += "Time Left: [time_remaining] s<hr><ul>"
-		for(var/i = 1 to length(choices))
-			var/votes = choices[choices[i]]
-			if(!votes)
-				votes = 0
-			. += "<li><a href='?_src_=vote;vote=[i]'>[choices[i]]</a> ([votes] votes)</li>"
-		. += "</ul><hr>"
-		if(admin)
-			. += "(<a href='?_src_=vote;vote=cancel'>Cancel Vote</a>) "
-	else
-		. += "<h2>Start a vote:</h2><hr><ul><li>"
-
-		var/avr = CONFIG_GET(flag/allow_vote_restart)
-		if(avr || admin)
-			. += "<a href='?_src_=vote;vote=restart'>Restart</a>"
-		else
-			. += "<font color='grey'>Restart (Disallowed)</font>"
-		if(admin)
-			. += "[FOURSPACES](<a href='?_src_=vote;vote=toggle_restart'>[avr ? "Allowed" : "Disallowed"]</a>)"
-		. += "</li><li>"
-
-		var/avm = CONFIG_GET(flag/allow_vote_mode)
-		if((length(GLOB.clients) > CONFIG_GET(number/maximum_clients_for_gamemode_vote)))
-			avm = FALSE
-		if(avm || admin)
-			. += "<a href='?_src_=vote;vote=gamemode'>GameMode</a>"
-		else
-			. += "<font color='grey'>GameMode (Disallowed)</font>"
-
-		if(admin)
-			. += "[FOURSPACES](<a href='?_src_=vote;vote=toggle_gamemode'>[avm ? "Allowed" : "Disallowed"]</a>)"
-
-		. += "</li><hr>"
-
-		if(admin)
-			. += "<li><a href='?_src_=vote;vote=groundmap'>Ground Map Vote</a></li>"
-			. += "<li><a href='?_src_=vote;vote=shipmap'>Ship Map Vote</a></li>"
-			. += "<li><a href='?_src_=vote;vote=custom'>Custom</a></li>"
-		. += "</ul><hr>"
-	. += "<a href='?_src_=vote;vote=close'>Close</a>"
-
-
-/datum/controller/subsystem/vote/can_interact(mob/user)
-	return TRUE
-
-
-/datum/controller/subsystem/vote/Topic(href, list/href_list, hsrc)
+/mob/verb/vote()
+	set category = "OOC"
+	set name = "Vote"
+	SSvote.ui_interact(usr)
+
+/datum/controller/subsystem/vote/ui_state()
+	return GLOB.always_state
+
+/datum/controller/subsystem/vote/ui_interact(mob/user, datum/tgui/ui)
+	// Tracks who is voting
+	if(!(user.client?.ckey in voting))
+		voting += user.client?.ckey
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Vote")
+		ui.open()
+
+/datum/controller/subsystem/vote/ui_data(mob/user)
+	var/list/data = list(
+		"allow_vote_groundmap" = CONFIG_GET(flag/allow_vote_groundmap),
+		"allow_vote_shipmap" = CONFIG_GET(flag/allow_vote_shipmap),
+		"allow_vote_mode" = CONFIG_GET(flag/allow_vote_mode),
+		"allow_vote_restart" = CONFIG_GET(flag/allow_vote_restart),
```

Comment:
These can be static data

Reply:
Can they though? since you can apparently set them using this UI

Reply:
when set call update static ui data